### PR TITLE
Change create items to multipart

### PR DESCRIPTION
**Pull request by jofmi**
# Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Description

Create items endpoint now behaves like update items (multipart form)

# Checklist:

- [ ] I have commented my code (or ChatGPT did), particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings, neither in my IDE nor in my browser
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by lebe1** [comment]
@almoxxx can you write the needed changes in the frontend to have a multipart/form call to create an item instead of a application/json call? :)
After that both branches can be tested well

**Comment by almoxxx** [comment]
> @almoxxx can you write the needed changes in the frontend to have a multipart/form call to create an item instead of a application/json call? :)

done in a new branch in frontend - pls review if ok :)

**Review by lebe1 — APPROVED** [review]

**Review by lebe1 — COMMENTED** [review]
I assumed the problem not uploading an image while creating an item must be in the frontend and therefore approved this PR first but after logging several messages and only receiving the first ones, I figured the problem lies still in the backend.

I feel like the CreateSimpleItem() delivers the positive return value for the handler in handlers.go:449 and therefore no image is uplaoded afterwards. Maybe thats the problem since I only receive the log message until SimpleCreateItem called?

P.S. I committed my log messages for you and you can test via the current main frontend branch by creating an item

image

**Comment by jofmi** [comment]
Hi, the CreateItemSimple is only a fallback option if there is an error with parsing the multipart form. The problem was that you did not see the error, which is likely still on the frontend. I added a line to log the error.

**Comment by jofmi** [comment]
We might consider to delete CreateItemSimple so that there is only one way to update, which is per multipart.

**Review by lebe1 — APPROVED** [review]
Thanks for adding the log message! That makes it way easier to find the real issue and as you assumed, I think as well that the final error lies within the frontend code [not providing a boundary in the multipart/form-data request](https://stackoverflow.com/questions/3508338/what-is-the-boundary-in-multipart-form-data).
I will create an issue for the frontend.
For now, I think we're good to go to stick with the function `CreateItemSimple()` since we would not be able to create an item right now....

Screenshot from 2023-12-13 16-12-50

File: app/handlers/handlers.go [hunk 1]
```diff
@@ -444,12 +462,48 @@ func ListItemsBackoffice(w http.ResponseWriter, r *http.Request) {
 //		@Security		KeycloakAuth
 //		@Router			/items/ [post]
 func CreateItem(w http.ResponseWriter, r *http.Request) {
-	var item database.Item
-	err := utils.ReadJSON(w, r, &item)
+	// Read multipart form
+	err := r.ParseMultipartForm(32 << 20)
```
Comment (nanu-c) [inline app/handlers/handlers.go:466]: Here I get the following error:

```
2023-12-13T16:29:51.389Z        error   1 error(s) decoding:
* 'ID' expected type 'int', got unconvertible type 'string', value: '0'
```

**Comment by nanu-c** [comment]
The multipart error should send an error 500 to the client and here https://github.com/augustin-wien/augustina-frontend/pull/94 is a frontend branch that corrects the multipart form post, which leads to new problems like:

```
2023-12-13T16:29:51.389Z        error   1 error(s) decoding:
* 'ID' expected type 'int', got unconvertible type 'string', value: '0'
```

File: app/handlers/handlers.go (hunk 1 above)
Comment (nanu-c) [reply app/handlers/handlers.go:466]: So it enters the if with for an unintended reason because it's still multipart.

**Comment by jofmi** [comment]
Hi, I tested with the frontend, and creating an item works apart from uploading an image!

**Review by lebe1 — APPROVED** [review]

**Pull request merged**